### PR TITLE
feat: analytics sdk integration on messaging

### PR DESCRIPTION
**Pull request by bozzelliandrea**
### Ticket:

- [17874](https://dev.azure.com/OGCIO-Digital-Services/Digital%20Services%20Programme/_workitems/edit/17874)

### Description

Integration of analytics inside messaging application

merge after: https://github.com/ogcio/life-events/pull/1028

## Type

- [X] **Dependency upgrade**
- [ ] **Bug fix**
- [X] **New feature**
- [ ] **Dev change**
- [ ] **Additional tests**
- [ ] **Documentation**
- [ ] **Other**

### Checklist:

- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] This change might impact the developer experience of others and should be communicated

### Screenshots:

N/A

### CODE REVIEW

File: apps/messaging-api/.env.sample [hunk 1]
```diff
@@ -19,4 +19,12 @@ LOGTO_M2M_SCHEDULER_APP_ID=bjd8knt92hxslvrfyezp7
...
 LOGTO_M2M_UPLOADER_APP_ID=upl1wqp66oisjwcjyder9
-LOGTO_M2M_UPLOADER_APP_SECRET=uploader_local_secret
\ No newline at end of file
+LOGTO_M2M_UPLOADER_APP_SECRET=uploader_local_secret
+# Analytics
+LOGTO_M2M_ANALYTICS_APP_SECRET=m2m_analytics_api_local_secret
+LOGTO_M2M_ANALYTICS_APP_ID=03emabp0dwznxw414mxme
+LOGTO_M2M_ANALYTICS_ORGANIZATION_ID=ogcio
+LOGTO_M2M_ANALYTICS_SCOPES=
+ANALYTICS_WEBSITE_ID=website_id
```
Comment (andreacappadona17) [inline apps/messaging-api/.env.sample:28]:
```suggestion
ANALYTICS_WEBSITE_ID=3e96e2f2-04a3-5576-e168-752846f4515b
```

File: apps/messaging/.env.sample [hunk 2]
```diff
@@ -23,4 +23,17 @@ PROFILE_BACKEND_URL=http://localhost:8003
...
+LOGTO_M2M_ANALYTICS_APP_SECRET=m2m_analytics_api_local_secret
+LOGTO_M2M_ANALYTICS_APP_ID=03emabp0dwznxw414mxme
+LOGTO_M2M_ANALYTICS_ORGANIZATION_ID=ogcio
+LOGTO_M2M_ANALYTICS_SCOPES=
+ANALYTICS_WEBSITE_ID=website_id
+ANALYTICS_URL=http://localhost:8075
+ANALYTICS_MATOMO_TOKEN=
+
+NEXT_PUBLIC_ANALYTICS_URL=http://localhost:8075
+NEXT_PUBLIC_ANALYTICS_WEBSITE_ID=04429483-e24e-9f55-96a5-675ff3eee9bc
```
Comment (andreacappadona17) [inline apps/messaging/.env.sample:38]:
```suggestion
NEXT_PUBLIC_ANALYTICS_WEBSITE_ID=3e96e2f2-04a3-5576-e168-752846f4515b
```

File: apps/messaging-api/.env.sample [hunk 3]
```diff
@@ -19,4 +19,13 @@ LOGTO_M2M_SCHEDULER_APP_ID=bjd8knt92hxslvrfyezp7
...
 UPLOAD_BACKEND_URL=http://localhost:8008
 LOGTO_M2M_UPLOADER_APP_ID=upl1wqp66oisjwcjyder9
-LOGTO_M2M_UPLOADER_APP_SECRET=uploader_local_secret
\ No newline at end of file
+LOGTO_M2M_UPLOADER_APP_SECRET=uploader_local_secret
+# Analytics
+LOGTO_M2M_ANALYTICS_APP_SECRET=m2m_analytics_api_local_secret
+LOGTO_M2M_ANALYTICS_APP_ID=03emabp0dwznxw414mxme
+LOGTO_M2M_ANALYTICS_ORGANIZATION_ID=ogcio
+LOGTO_M2M_ANALYTICS_SCOPES=
```
Comment (andreacappadona17) [inline apps/messaging-api/.env.sample:27]:
```suggestion
LOGTO_M2M_ANALYTICS_SCOPES=analytics:website:tracking
```

File: apps/messaging/.env.sample [hunk 4]
```diff
@@ -23,4 +23,19 @@ PROFILE_BACKEND_URL=http://localhost:8003
...
-LOG_LEVEL=debug
\ No newline at end of file
+LOG_LEVEL=debug
+# Analytics
+LOGTO_OIDC_ENDPOINT=http://localhost:3301/oidc
+LOGTO_M2M_ANALYTICS_APP_SECRET=m2m_analytics_api_local_secret
+LOGTO_M2M_ANALYTICS_APP_ID=03emabp0dwznxw414mxme
+LOGTO_M2M_ANALYTICS_ORGANIZATION_ID=ogcio
+LOGTO_M2M_ANALYTICS_SCOPES=
+ANALYTICS_WEBSITE_ID=website_id
```
Comment (andreacappadona17) [inline apps/messaging/.env.sample:33]:
```suggestion
ANALYTICS_WEBSITE_ID=3e96e2f2-04a3-5576-e168-752846f4515b
```

File: apps/messaging/.env.sample [hunk 5]
```diff
@@ -23,4 +23,19 @@ PROFILE_BACKEND_URL=http://localhost:8003
...
 LOGTO_M2M_UPLOADER_APP_SECRET=uploader_local_secret
-LOG_LEVEL=debug
\ No newline at end of file
+LOG_LEVEL=debug
+# Analytics
+LOGTO_OIDC_ENDPOINT=http://localhost:3301/oidc
+LOGTO_M2M_ANALYTICS_APP_SECRET=m2m_analytics_api_local_secret
+LOGTO_M2M_ANALYTICS_APP_ID=03emabp0dwznxw414mxme
+LOGTO_M2M_ANALYTICS_ORGANIZATION_ID=ogcio
+LOGTO_M2M_ANALYTICS_SCOPES=
```
Comment (andreacappadona17) [inline apps/messaging/.env.sample:32]:
```suggestion
LOGTO_M2M_ANALYTICS_SCOPES=analytics:website:tracking
```

Comment (bozzelliandrea) [reply apps/messaging/.env.sample]: fixed! (all comments)

**Review by alewin — APPROVED** [review]

**Pull request merged**